keps/sig-instrumentation/1668-trace-popagating/README.md

```diff
@@ -0,0 +1,501 @@
+  per day <= X". It's impossible to provide comprehensive guidance, but at the very
+  high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99,9% of /health requests per day finish with 200 code
+
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?**
+  Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
+  implementation difficulties, etc.).
+
+### Dependencies
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **Does this feature depend on any specific services running in the cluster?**
+  Think about both cluster-level services (e.g. metrics-server) as well
+  as node-level agents (e.g. specific version of CRI). Focus on external or
+  optional services that are needed. For example, if this feature depends on
+  a cloud provider API, or upon an external software-defined storage or network
+  control plane.
+
+  For each of these, fill in the following—thinking about running existing user workloads
+  and creating new ones, as well as about cluster-level services (e.g. DNS):
+  - [Dependency name]
+    - Usage description:
+      - Impact of its outage on the feature:
+      - Impact of its degraded performance or high-error rates on the feature:
+
+
+### Scalability
+
+_For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them._
+
+_For beta, this section is required: reviewers must answer these questions._
+
+_For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field._
+
+* **Will enabling / using this feature result in any new API calls?**
+  N/A
+
+* **Will enabling / using this feature result in introducing new API types?**
+  N/A
+
+* **Will enabling / using this feature result in any new calls to the cloud 
+provider?**
+  N/A
+
+* **Will enabling / using this feature result in increasing size or count of 
+the existing API objects?**
+  N/A
+
+* **Will enabling / using this feature result in increasing time taken by any 
+operations covered by [existing SLIs/SLOs]?**
+  N/A
+
+* **Will enabling / using this feature result in non-negligible increase of 
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+  TBD
+
+### Troubleshooting
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How does this feature react if the API server and/or etcd is unavailable?**
+  The feature will be unavailable.
```

Comment:
Is this true? If this uses metadata from objects that are already cached, would this be entirely unavailable?

Reply:
About this question, our thought are as following:
The trace context propagation is a series of steps:

1. requests with/without trace context reach apiserver
2. apiserver create/update trace context
3. requests with trace context reach webhook
4. webhook persists trace context to etcd as object's annotation
5. in controllers, extract context from one object to propagate to another object.
6. controllers send requests with trace context to apiserver
7. back to 1
If apiserver and/or etcd is unavailable, all of the steps except for 5 will be unavailable.

For 5, even though the metadata from objects that are already cached can be propagated to another object, the trace information will not be persisted eventually. So the feature actually is unavailable.